src/NHibernate/Criterion/PropertyExpression.cs

```diff
@@ -70,25 +70,25 @@ protected PropertyExpression(string lhsPropertyName, IProjection rhsProjection)
 
 		public override SqlString ToSqlString(ICriteria criteria, ICriteriaQuery criteriaQuery)
 		{
-			SqlString[] columnNames =
-				CriterionUtil.GetColumnNames(_lhsPropertyName, _lhsProjection, criteriaQuery, criteria);
-			SqlString[] otherColumnNames =
-				CriterionUtil.GetColumnNames(_rhsPropertyName, _rhsProjection, criteriaQuery, criteria);
+			var columnNames =
+				CriterionUtil.GetColumnNamesAsSqlStringParts(_lhsPropertyName, _lhsProjection, criteriaQuery, criteria);
+			var otherColumnNames =
+				CriterionUtil.GetColumnNamesAsSqlStringParts(_rhsPropertyName, _rhsProjection, criteriaQuery, criteria);
 
 			SqlStringBuilder sb = new SqlStringBuilder();
 			if (columnNames.Length > 1)
 			{
 				sb.Add(StringHelper.OpenParen);
 			}
 			bool first = true;
-			foreach (SqlString sqlString in SqlStringHelper.Add(columnNames, Op, otherColumnNames))
+			for (var i = 0; i < columnNames.Length; i++)
```

Comment:
Since this loop is rewritten anyway, why not use the `if (columnNames.Length > 1)` above to handle the first case then iterate from `1` and remove the `first` bloat?

Reply:
Done